tests/test_models/test_tbats.py

```diff
@@ -258,16 +324,23 @@ def test_components_names(periodic_ts, estimator, params, components_names):
         TBATSModel,
     ),
 )
-def test_seasonal_components_not_fitted(small_periodic_ts, estimator):
+@pytest.mark.parametrize("method,use_future", (("predict_components", False), ("forecast_components", True)))
+def test_seasonal_components_not_fitted(small_periodic_ts, estimator, method, use_future):
     model = estimator(seasonal_periods=[7, 14], use_arma_errors=False)
     model.fit(small_periodic_ts)
 
-    future = small_periodic_ts.make_future(3).to_pandas(flatten=True)
     segment_model = model._models["segment_1"]
     segment_model._fitted_model.params.components.seasonal_periods = []
 
+    pred_ts = small_periodic_ts
```

Comment:
This pattern occurs in several tests, do we really need it? I guess it makes tests overcomplicated. Better to test all the methods on appropriate fixtures in long format and use adapters instead of models

Reply:
We can create separate fixtures for train and test, do't generate them inside the tests with make_future

Reply:
This pattern is a result of parametrization with `forecast_decompose` and `predict_decompose`. Now tests updated and it looks a bit different.